src/main/java/io/nitric/faas/Response.java

```diff
@@ -0,0 +1,102 @@
+package io.nitric.faas;
+
+/*-
+ * #%L
+ * Nitric Java SDK
+ * %%
+ * Copyright (C) 2021 Nitric Technologies Pty Ltd
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import com.google.protobuf.ByteString;
+import io.nitric.proto.faas.v1.TriggerResponse;
+import io.nitric.proto.faas.v1.TopicResponseContext;
+import io.nitric.proto.faas.v1.HttpResponseContext;
+
+/**
+ * Response model for a NitricFunction
+ */
+public class Response {
+    private byte[] data;
+    private final ResponseContext context;
+
+    /**
+     * Create a new Nitric Response
+     *
+     * @param data The data for the response
+     * @param context The context of the response
+     */
+    protected Response(byte[] data, ResponseContext context) {
+        this.data = data;
+        this.context = context;
+    }
+
+    /**
+     * Retrieve the data contained in the response
+     *
+     * @return The response data as bytes
+     */
+    public byte[] getData() {
+        return this.data;
+    }
+
+    /**
+     * Set the data for this response
+     *
+     * @param data The data as an array of bytes
+     */
+    public void setData(byte[] data) {
```

Comment:
This was more about giving the user full control over a response object. I think immutability in this case is ideal, my only thought was if the use for whatever reason wanted to change their return data based on the context of the Response.

```java
var response = trigger.buildResponse();

if (response.getContext().isHttp()) {
  // Write my response as HTML if the trigger was raised by a HTTP request
  response.getContext().asHttp().addHeader("Content-Type", "text/html");
  response.setData(''<html><body>Successfully Processed Request!</body></html>".getBytes());
} else {
 // Otherwise just write a single string
  response.setData("Successfully processed request!".getBytes());
}

return response;
```